src/components/shared/RightSidebar/SocialLinks.tsx

```diff
@@ -1,52 +1,175 @@
 "use client";
 import Button from "@/components/interface/Button";
-import Image from "next/image";
-import React from "react";
+import { Separator } from "@/components/ui/separator";
+import React, { useEffect, useState } from "react";
 import IconLink from "../LeftSidebar/IconLink";
 import { usePathname } from "next/navigation";
+import { getUserSocialLinks } from "~/lib/actions/users";
+import { SocialMediaLink } from "~/types";
+import { handleFieldArrayError } from "@/components/shared/utils/FormErrorHandlers";
 
-const SocialLinks = () => {
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { SocialLinksSchema } from "~/lib/validators/socialLinks.schema";
+import SocialLinkInput from "./SocialLinkInput";
+import { updateUserSocialLinks } from "~/lib/actions/users";
+
+interface SocialLinksProps {
+  userID: string;
+}
+
+const SocialLinks: React.FC<SocialLinksProps> = ({ userID }) => {
   const pathname = usePathname();
-  if (!pathname.includes("profile")) return null;
-  const socialLinks = [
-    {
-      title: "LinkedIn",
-      icon: "icons/github.svg",
-      link: "https://www.linkedin.com/in/username",
-    },
-    {
-      title: "GitHub",
-      icon: "icons/github.svg",
-      link: "https://www.linkedin.com/in/username",
+
+  const [socialLinks, setSocialLinks] = useState<SocialMediaLink[]>();
+
+  const {
+    register,
+    control,
+    handleSubmit,
+    formState: { errors },
+    watch,
+    reset,
+  } = useForm({
+    resolver: zodResolver(SocialLinksSchema),
+    defaultValues: {
+      socialLinks: socialLinks,
     },
+  });
+
+  useEffect(() => {
+    const getTheLinks = async () => {
+      const socialLinks = await getUserSocialLinks(userID);
+      setSocialLinks(socialLinks);
+      reset({
+        socialLinks: socialLinks.map((link: any) => {
+          return { url: link.url, site: link.site, username: link.username };
+        }),
+      });
+    };
+    getTheLinks();
+  }, [userID, reset]);
+
+  function getIcon(site: string) {
+    switch (site) {
+      case "twitter":
+        return "twitter";
+      case "x":
+        return "x";
+      case "instagram":
+        return "instagram";
+      case "facebook":
+        return "facebook";
+      case "linkedin":
+        return "linkedin";
+      case "github":
+        return "github";
+      case "website":
+        return "website";
+      default:
+        return "website";
+    }
+  }
+
+  const sites = [
+    "twitter",
+    "instagram",
+    "facebook",
+    "linkedin",
+    "github",
+    "website",
   ];
```

Comment:
```suggestion
  const sites = [
    "twitter",
    "instagram",
    "facebook",
    "linkedin",
    "github",
    "website",
  ];
  function getIcon(site: string) {
    if (sites.includes(site)) return site;
    return "website";
  }
```
If we just have a hardcoded list- maybe we can simplify this to just take advantage of the 'sites' array.  I've also removed "x" as an option as that seems to only have been referenced in the `getIcon` switch but nowhere else.

Correct me if I am wrong on that.